Source/Core/Common/FileUtil.cpp

```diff
@@ -483,8 +483,18 @@ bool CreateEmptyFile(const std::string& filename)
 }
 
 // Recursive or non-recursive list of files and directories under directory.
-FSTEntry ScanDirectoryTree(const std::string& directory, bool recursive)
+FSTEntry ScanDirectoryTree(const std::string& input_directory, bool recursive)
 {
+  std::string directory = input_directory;
```

Comment:
If you always need to copy the string, it's better to not use a reference to begin with:

```suggestion
FSTEntry ScanDirectoryTree(std::string directory, bool recursive)
{
```

(The main advantage is when the argument is an xvalue – in that case this lets you skip the copy.)

Reply:
Fair point.